OEP 62: Working Groups

kdmccormick opened the pull request:
**Reviewers**: Before I find an arbiter and moving this into proper review, and I am looking for some "pre-review". That is, I am looking for your gut reactions, high-level opinions, etc. Is this the structure you envision for WGs, or is it completely different? Do we need an OEP for it at all? And is now the right time? etc.

Don't sweat the formatting or typos right now. Many links are broken or missing. I'm aware, and I plan to fix them all later on.

**Timeline**: I'm pushing this forward as much as I have time to, right now. As far as I can tell, this OEP is not blocking anything. I think putting it up for proper Review before the end of January would be nice, though.

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+
+Since then, several more working groups have organized themselves using a similar model. The groups operate autonomously, but have largely standardized around a set of best practices to foster an inclusive and productive environment. Many of the groups play important roles in the project, such as organizing community events, developing standard user interface components, and driving project strat ...
+
+Given the importance of working groups to the health of the Open edX project, it may surprise new Open edX contributors that litle formal documentation exists to establish the working group system. The only official guidance so far is found in the `Technical Oversight Committee (TOC) Charter`_:
```
```suggestion
Given the importance of working groups to the health of the Open edX project, it may surprise new Open edX contributors that little formal documentation exists to establish the working group system. The only official guidance so far is found in the `Technical Oversight Committee (TOC) Charter`_:
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+
+    b. The TOC will typically delegate tactical decision making to those who best understand the details, including Working Groups, Task Forces, Expert Groups, etc. Only significant changes in the status quo or decisions that are difficult to reverse should be escalated to the TOC for consideration
+
+While this excert is true, TOC-created working groups are only a special case of the broader working group system. In this document, we aim to capture the generally-applicable processes and best practices for working groups so that they can be more readily understood by newcomers to the Open edX project and more easily referenced by all community members. In order to avoid disruptive the already- ...
```
```suggestion
While this excerpt is true, TOC-created working groups are only a special case of the broader working group system. In this document, we aim to capture the generally-applicable processes and best practices for working groups so that they can be more readily understood by newcomers to the Open edX project and more easily referenced by all community members. In order to avoid disrupting the already-successful working group system that exists today, this OEP mandates only a lightweight framework, one which mostly aligns with how the community already operates.
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+Name and Purpose
+----------------
+
+Evey working group must have a specific name and statement of pupose. The purpose can be brief: a sentance or a paragraph is sufficient. The name and purpose of can be amended at any time via consensus of working group members, although major changes should be announced to the community.
```
```suggestion
Evey working group must have a specific name and statement of purpose. The purpose can be brief: a sentence or a paragraph is sufficient. The name and purpose can be amended at any time via consensus of working group members, although major changes should be announced to the community.
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+* any information that will help their successor transition into the role, and
+* optionally: any person or people who they recommend as their successor.
+
+The working group should aim to fill the role before the departing chairperson steps down. If they cannot, the group is encouraged to reach out the `Core Contributors`_ group for assisstance.
```
```suggestion
The working group should aim to fill the role before the departing chairperson steps down. If they cannot, the group is encouraged to reach out the `Core Contributors`_ group for assistance.
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+* Collaborate using publicly-visible communication channels and/or task trackers.
+* Generate and review work artifacts (code, documents, images, decisions, etc.) in publicly-visible spaces.
+* Publicize any and all meeting times in UTC, with an open invite to Open edX community members.
+* Maintain an up-to-date listing of group metadata, including:
```
Is it worth codifying where this listing should be written up?  For discoverability?

kdmccormick replied:
Yeah, good idea, I'll refactor this section a bit in my next draft to point to a particular page.

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+
+Such groups should otherwise follow the openness & inclusivity guidelines to their best effort.
+
+Finally, as with any activity in the Open edX community, behavior in working groups that conflicts with the `Open edX Code of Conduct`_ should be swiftly by the means detailed in that policy.
```
Swiftly... dealt with? 🤔  Sounds sinister. 😂

kdmccormick replied:
whoops, swiftly _addressed_.

Not meant to be sinister 😁 Just trying to be clear: although groups need to be inclusive, they don't need to tolerate jerkish behavior. In fact, to be most inclusive, they should address jerkish behavior as soon as possible. Happy to reword.

kdmccormick replied:
Reworded this a bit.

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+* assist community members; or
+* simply discuss an aspect Open edX platform.
+
+Certain working group may step up to fulfill a responsibility called for in an Open edX Propsal. For example, at the time of writing, the Product Working Group is leading the implemenation `the Core Product from OEP-57`_, the DevOps Working Group handles `the release process from OEP-10`_, and the Architecture Coordination Working Group facilitates `the architecture advisory process from OEP-56`_ ...
```
```suggestion
Certain working groups may step up to fulfill a responsibility called for in an Open edX Proposal. For example, at the time of writing, the Product Working Group is leading the implementation `the Core Product from OEP-57`_, the DevOps Working Group handles `the release process from OEP-10`_, and the Architecture Coordination Working Group facilitates `the architecture advisory process from OEP-56`_. Due to the fluid nature of working groups, the association between OEP-declared responsibilities and specific working group names is subject to change over time.
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+
+Certain working group may step up to fulfill a responsibility called for in an Open edX Propsal. For example, at the time of writing, the Product Working Group is leading the implemenation `the Core Product from OEP-57`_, the DevOps Working Group handles `the release process from OEP-10`_, and the Architecture Coordination Working Group facilitates `the architecture advisory process from OEP-56`_ ...
+
+Furthermore, working groups can step up to support maintenance Open edX components, per the process detailed in `OEP-55 <https://docs.openedx.org/projects/openedx-proposals/en/latest/processes/oep-0055-proc-project-maintainers.html>`_.
```
```suggestion
Furthermore, working groups can step up to support maintenance of Open edX components, per the process detailed in `OEP-55 <https://docs.openedx.org/projects/openedx-proposals/en/latest/processes/oep-0055-proc-project-maintainers.html>`_.
```

davidjoy commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+
+Finally, one of the remaining working group members should update the public listing working groups and announce the dissolution and any related responsibility transfers to the community.
+
+If there seem to be no remaining members in working group and the last chairperson cannot be reached, then the group is *de facto* dissolved. Any Core Contributor is enocuraged to take the above steps in lieu of the abandoned working group.
```
```suggestion
If there seem to be no remaining members in working group and the last chairperson cannot be reached, then the group is *de facto* dissolved. Any Core Contributor is encouraged to take the above steps in lieu of the abandoned working group.
```

davidjoy reviewed (commented):
I didn't show up intending to proofread, but sorta did.  Hope you don't mind!

Content-wise, this is nice to see all in one place.  One thing I might want to add is something about the importance of documenting a WG's goals and outcomes somewhere.  There are benefits to doing so - it'll help the group measure their own success, celebrate it, adjust when they're not quite reaching their stated goals, and (potentially) know when the group has accomplished its mission and should be (happily) dissolved.

nedbat commented:
@kdmccormick let us know when you are ready for us to review... (and spell-check!)

kdmccormick commented:
Will do!

kdmccormick commented:
To be clear, all comments are welcome, although I cannot promise when I'll get to responding to them. I'll notify when I'm ready to actively iterate on review.

kdmccormick commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,191 @@
+Background & Motivation
+***********************
+
+The first community-lead Open edX working groups were the *Transifex Working Group* and the *Build-Test-Release Working Group*. The former group took responsibility for creating and curating translations of strings on the Open edX platform; the latter took responsility for the platform's named releases. These groups were formed by community members who saw a need in the project and took the initi ...
```
Reviewers: I wrote this to the best of my knowledge, but I am not 100%  these two groups were first. Would love a fact-check on this.

kdmccormick commented:
I updated the description:
> Reviewers: Before I find an arbiter and moving this into proper review, and I am looking for some "pre-review". That is, I am looking for your gut reactions, high-level opinions, etc. Is this the structure you envision for WGs, or is it completely different? Do we need an OEP for it at all? And is now the right time? etc.

Anyone following this PR is welcome to pre-review!

hurtstotouchfire commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,218 @@
+Forming
+-------
+
+To form a working group, any community member can share an announcement describing:
```
Do you want to specify where to announce? I think ideally that would be able to reference another doc about "how to announce things to the whole community", which may already exist?

kdmccormick replied:
A confluence doc is a good idea. That way, if we change how we announce things, we won't need to remember to come back and update this section.

hurtstotouchfire commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,218 @@
+* how to get involved in the new group, and
+* optionally: who they suggest as the initial chairperson.
+
+If the announcement generates sufficient interest, then the working group is formed. If not already selected, their first order of business should be to select a chairperson. Once selected, the new chairperson should publicise the new group's details as described above.
```
> If the announcement generates sufficient interest

This seems very vague and I don't think it's correct either. I would say "if the announcement generates no objections" or similar instead, and maybe even indicate what to do if there are objections (I would escalate to tCRIL).

kdmccormick replied:
Right, I was being purposefully a little vague here. I am imagining "sufficient interest" to mean "you got enough interest that moving forward seems worthwhile."

> if the announcement generates no objections

Hm, this is interesting. Do we folks to be able to block the creation of a working group by objecting to it? I'm having trouble imagining a scenario (excluding obvious code of conduct violations) that would warrant objection. Do you have any situations in mind?

hurtstotouchfire commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,218 @@
+
+Without an OEP, working groups would still continue operating as they are today, possibly without any issues. So, we could have forgone this OEP.
+
+However, we believe that when critical processes left undocumented, it becomes harder for newcomers to aquaint themselves with a project. Furthermore, undocumented proccesses work because certain community members happen to remember them; as those community members retire or move on to other projects, their knowledge can be lost. Finally, this OEP does add certain new guidelines that we believe a ...
```
Yes. This was extremely confusing to approach as a newer person in the community. I definitely think it needs to be more explicit.

kdmccormick replied:
That is great feedback to hear, thank you. I was worried writing this that I might just be stating the obvious, to the benefit of nobody.

hurtstotouchfire commented on oeps/processes/oep-0062-proc-working-groups.rst:
```diff
@@ -0,0 +1,218 @@
+
+We could define a stricter framework for working groups. For example, we could require that new working groups be approved by some set of Open edX leaders.
+
+TODO: Add some more justification on why we didn't make a stricter framework.
```
I think this should be an attempt to make the status quo explicit, and that iterations may follow. Right now, trying to make things more strict might break implicit systems or benefits that we don't even know exist.

kdmccormick replied:
Agreed.

sarina commented:
@kdmccormick - do you feel there is still a need for this document?

kdmccormick commented:
@sarina Yeah, I do.

sarina commented:
Do you think there's someone else in the community that could take the charge on getting this over the line? I don't see this as the most valuable OEP we're currently focusing on

kdmccormick commented:
Yeah, this isn't high priority, and hasn't felt high priority since late 2022, hence the lack of movement. Nobody comes to mind in particular, although I'm sure plenty of people could get this landed if they were interested.

When I started this OEP, we didn't have anything written down for what working groups are, who is allowed to join one, who is allowed to make one. Well, we had one document, and it literally said "if you want to start a working group, ask Ed and Braden for permission" 😛  We also had WG wiki pages all over the place, some groups with calendar events and some without,  some with public Slack channels and some without, some with public boards and some with private boards, etc. I started this OEP because it felt like a big mess, one which Axim didn't acutely feel because we tend to just know what's going on at all times, but one that new community members would absolutely feel when they try to join a meeting for a working group that quietly disbanded a year ago.

In the end, I didn't have time to push the OEP through, so I [wrote this page](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3621519401/How+to+Make+a+New+Working+Group), fixed up the wiki, made sure everyone had a Slack channel with a matching name and a public calendar invite, and moved on. IIRC you also helped a bunch with that @sarina , particularly that part about getting people onto public GH issue boards.

So, maybe we're good, as long as Axim regularly goes though and makes sure that everything is reasonably tidy, there are no "active" working groups that have actually disappeared, etc.

So that's the concrete reason I started this. Abstractly, though, the question of whether we need this OEP hinges on the question "what are the purpose of OEPs?" Is it:

1. to record the big decisions that the community explicitly made? or
2. to have a collection of principles under which the project operates?

If it's (1), then I'll accept the argument that we don't need this OEP because WGs happened organically. If it's (2), though, then I'd assert we definitely need this OEP, since working groups are just important to the functioning community as core contributors, deprecation, maintenance, and named releases.

sarina commented:
`/me reads OEP-1`

Yeah it's not super clear to me! The preamble states,

> An Open edX Proposal specifies a best practice, architectural decision, or community process that the Open edX community has agreed should be adopted by the project.

But the opening paragraph doesn't even make an affordance for non-technical OEPs at all:

> OEP (pronounced “oh-epp”) stands for Open edX (Enhancement) Proposal. An OEP is a document that details a specific technology decision being made by the Open edX community, in the form of a best practice, architecture design, or process adjustment. An OEP should provide the use cases and rationales that surround that choice. OEPs are not the only way for a change to be made to Open edX, however. The goal is to create a collection of OEP documents as a repository or knowledge archive of architectural choices made for the platform.

Honestly I think I lean to this doesn't need an OEP. The process OEPs that stand out - Core Committers and Maintainers - describe a process that on review was a bit contentious and really needed an agreed upon blueprint/set of requirements for us to move forward and collaboratively build the programs they specified. I feel with Working Groups, between a robust set of wiki pages and the fairly disparate ways in which working groups act, it feels more community-self-organizing and determination over a program that needs a clear blueprint with which to operate. But, it's an interesting and unclear question and I think I'd like to invite team members who are interested to weigh in. And if we choose to not pursue this as an OEP, we should post to the forums with a link to this thread for anyone else who wants to discuss.

sarina commented:
We discussed this with the Axim Eng team, and we don't feel there is a pressing need to codify this. If anyone disagrees, please feel free to fork this draft and open up a new PR.
